docs/docs/getting_started/tooling/noir_codegen.md

```diff
@@ -0,0 +1,135 @@
+---
+title: Noir Codegen for Typescript
+description: Learn how to use Noir codegen to generate Typescript bindings 
+keywords: [Nargo, Noir, compile, TypeScript]
+sidebar_position: 2
+---
+
+You can generate TypeScript bindings with `noir_codegen`, allowing for easier and faster integration of your Noir circuits into TypeScript projects.
+
+It is used in conjunction with `nargo export` which generates JSON for specified functions in a Noir library (not binary or contract circuits).
+
+## Installation
+
+You can install `noir_codegen` into your project using `yarn` or `npm`:
+
+```bash
+yarn add @noir-lang/noir_codegen
+```
+
+or 
+
+```bash
+npm install @noir-lang/noir_codegen
+```
+## Usage
+
+### Export ABI of specified functions
+
+Use the `#[export]` macro to indicate which functions in the noir library that you'd like to export.
+```rust
+#[export]
+fn your_function(...
+```
+
+The following command, when run from the directory with `Nargo.toml`, will create a folder with a .json file per exported function:
+
+```bash
+nargo export
+```
+The .json files live in a new `exports` directory.
+
+You can also specify the directory of Noir programs using `--program-dir` as follows:
+
+```bash
+nargo export --program-dir=./path/to/your/noir/program
+```
+
+### Run noir-codegen 
+
+The `noir-codegen` command, pass the path of your exported JSON file(s) (wildcard characters accepted too, `*.json`). :
+
+```bash
+yarn noir-codegen ./export/your_function.json
+```
+
+An `exports` directory is created with an index.ts file containing all exported functions.
+
+You can optionally specify an output dir for your Typescript bindings to live with `--out-dir`:
+
+```bash
+yarn noir-codegen ./export/*.json --out-dir ./path/to/output/dir
+```
+
+
+## Example
+
+For example, if you have a Noir program like this:
+
+```rust
+struct MyStruct {
+    foo: bool,
+    bar: [str<5>; 3],
+    baz: Field
+}
+
+struct NestedStruct {
+    foo: MyStruct,
+    bar: [MyStruct; 3],
+    baz: u64
+}
+#[export]
+fn exported_function_foo(x: u64, y: u64, array: [u8; 5], my_struct: NestedStruct, string: str<5>) -> (u64, u64, MyStruct) {
+    assert(array.len() == 5);
+    assert(my_struct.foo.foo);
+    assert(string == "12345");
+
+    print(x);
+    assert(x < y);
+    (x + y, 3, my_struct.foo)
+}
+
+fn unexported_function(x: u64, y: u64) {
+    assert(x != y);
+}
+```
+
+After following the usage instructions, you will have an `index.ts` that looks like this:
```

Comment:
```suggestion
You can run `nargo export` followed by `noir-codegen` (assuming you installed it globally), and you'll see an `index.ts` file similar to:
```

Reply:
I'll rephrase, but not lean into the global again.